File: 800/2175A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        int n;
        cin >> n;
        vector<int> a(n);

        for (int i = 0; i < n; i++)
            cin >> a[i];

        sort(a.begin(), a.end());

        int cnt = 0;
        for (int i = 1; i < n; i++) 
        {
            if (a[i] == a[i - 1]) 
                continue;

            cnt++;
        }
        cnt++; // include the first distinct element

        int ans = cnt;
        for (int i = 0; i < n; i++) 
        {
            if (a[i] < cnt) 
                continue;

            ans = a[i];
            break;
        }

        cout << ans << "\n";
    }

    return 0;
}

// solution 2
// #include <bits/stdc++.h>
// using namespace std;

// int main() 
// {
//     int t;
//     cin >> t;

//     while (t--) 
//     {
//         int n;
//         cin >> n;
//         unordered_map<int, int> mp;

//         for (int i = 0; i < n; i++) 
//         {
//             int x;
//             cin >> x;
//             mp[x]++;
//         }

//         int val = mp.size(); // number of distinct elements

//         while (mp.find(val) == mp.end())
//             val++;

//         cout << val << "\n";
//     }
//     return 0;
// }

File: 800/2175A/main.java
// solution 1

import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            int n = sc.nextInt();
            int a[] = new int[n];
            
            for(int i=0;i<n;i++)
                a[i] = sc.nextInt();
            
            Arrays.sort(a);
            int cnt = 0;
            
            for(int i=1;i<n;i++)
            {
                if(a[i] == a[i-1])
                    continue;
                
                cnt++;
            }
            
            cnt++;
            
            int ans = cnt;
            for(int i=0;i<n;i++)
            {
                if(a[i] < cnt)
                    continue;
                
                ans = a[i];
                break;
            }
            
            System.out.println(ans);
        }
    }
}

// solution 2
// import java.util.*;
// public class Main
// {
//     public static void main(String args[])
//     {
//         Scanner sc = new Scanner(System.in);
//         int t = sc.nextInt();
        
//         while(t-- > 0)
//         {
//             int n = sc.nextInt();
//             HashMap<Integer, Integer> mp = new HashMap<>();
            
//             for(int i=0;i<n;i++)
//             {
//                 int a = sc.nextInt();
//                 if(mp.get(a) == null)
//                     mp.put(a, 1);
                
//                 mp.put(a, mp.get(a)+1);
//             }
            
//             int val = mp.size();
            
//             while(mp.get(val) == null)
//                 val++;
            
//             System.out.println(val);
//         }
//     }
// }